Fixes/1.9 fixes - Made requested fixes to bestiary

RichPrune opened the pull request:
All pr changes done

Adds "some mob icons" rp i used textures for bestiary

Disabled toasts for all bestiary quests, made requested book quest/legend at the top give xp and autocomplete via temp, removed all unnecessary boss dependencies. Figured out the showing the quests before you get them upon world load is because of the same problem as the optional thing: if all tasks are optional it autocompletes upon world load. Added a tag task that isnt optional to every biome quest and tags in crafttweaker script for each category.

- adds more tags to inventory mod tags

- add beastiary questbook page

- replace questbook item with dormant eye

Jonah-Hansen commented on src/overrides/scripts/biome_tags.zs:
```diff
@@ -0,0 +1,133 @@
+<tag:worldgen/biome:chocolate:forests>.addId(<resource:minecraft:windswept_forest>);
+<tag:worldgen/biome:chocolate:forests>.addId(<resource:minecraft:birch_forest>);
+<tag:worldgen/biome:chocolate:forests>.addId(<resource:atmospheric:laurel_forest>);
+
```
forest tag missing vanilla forest? should probably do a double check on all of these

RichPrune replied:
ur right, i coulda sworn I added it, weird. I think this is a one off though because i used the ct dump biome list.

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2426,13 +2534,14 @@
 					}
-					id: "ftbquests:book"
+					id: "enigmaticlegacy:enigmatic_eye"
+					tag: { }
```
this change should not be implemented here. we can merge its pr in for 1.9 patch if you really want but it shouldnt be in this one

RichPrune replied:
oh this was for the dormant eye patch but it also got put in here because temp takes all the changes. You're right it shouldn't be implemented here, the solution seems to be in the future doing one thing at a time with each quest book page instead of like going back and deleting progress to submit to another branch or something. I think we could do the dormant eye change for the patch since its minor, if you dont mind

Jonah-Hansen commented on src/overrides/config/paxi/resourcepacks/choccompatv1.3/assets/enigmaticlegacy/lang/en_us.json:
this change should not be included in this pr

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2688,10 +2796,30 @@
+					team_reward: false
+					type: "command"
+				}
 			]
```
adding more biome particles? these should be their own pr, and ensure you add them to the settings page as well.

tbh we should not be adding more biome particles knowing how bad they are for performance. find a better way to implement them before adding more

Jonah-Hansen replied:
remove added particles. these cause significant performance issues. find a different way to do this if you want it to happen.

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2645,7 +2754,6 @@
 			can_repeat: true
 			dependencies: ["177CCA876BBC9A3E"]
-			disable_toast: true
 			id: "5B584CE502BC9124"
```
why no disable toast?

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2701,7 +2829,6 @@
 			can_repeat: true
 			dependencies: ["572E7C8B81EA10ED"]
-			disable_toast: true
 			id: "71205726B9342781"
```
why no disable toast?

RichPrune replied:
same here, also set to auto toast

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2725,7 +2852,6 @@
 			can_repeat: true
 			dependencies: ["5E0D0EFACA192EEB"]
-			disable_toast: true
 			id: "6FB30BCA8CBF9926"
```
why no disable toast?

RichPrune replied:
I dont remember editing this but it is set to auto no_toast so its fine

Jonah-Hansen replied:
pretty sure these are some of the quests giving toasts when first joining a world. please check in game.

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -2898,14 +3024,44 @@
+					team_reward: false
+					type: "command"
+				}
 			]
```
what does this do? it should be its own pr

RichPrune replied:
what are you referring to? You only clip half the thing I can't even find the quest youre talking about without a name

Jonah-Hansen replied:
Click on this file to see it in its entirety, then scroll down to this comment

RichPrune replied:
my fault og, deleted these

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/bestiary.snbt:
maybe we leave toasts on for the individual biome tasks but off for mobs and for the non-optional biome tag task? this way you get a little notification each time you enter a new biome for the first time? idk its your call

seems many of the animals tasks still give toasts. Im on the fence about this. for people who are trying to observe all the animals it would be nice to see when the "observe" has triggered, but also might get annoying. 

whatever you decide (or maybe open a dev poll?) make sure its consistent across all entries

RichPrune replied:
so it seems like when there are multiple animals it toasts the first ones you find but not the one that completes it. This is a reference to the fact that it hates me

RichPrune replied:
I cant do that, disabling the quest disables the tasks unless the thing i mentioned when there are multiple

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/bestiary.snbt:
also, opening the beastiery page starts the crosshair at some random place in the middle, so you have to drag around to find the top. not sure if theres something that can be done about that

RichPrune replied:
i dont think there is unforunately

Jonah-Hansen reviewed (changes requested):
should add a checkmark to this quest, so the quest is not marked as complete unless the player has read the description

![image](https://github.com/user-attachments/assets/1edecf9d-640f-4f9c-8504-f68cfa96f095)

leave this as no task with xp reward so it shows the "unclaimed reward" icon, encouraging players to view it

![image](https://github.com/user-attachments/assets/efac18ca-93d0-49ad-b4f3-3b91c6d8b19b)


add a checkmark task so players can "mark as read" 
![image](https://github.com/user-attachments/assets/e0611504-11bf-44d5-9477-b9cdbf39c806)


still some missing textures
![image](https://github.com/user-attachments/assets/f4d8b7e0-7fee-4043-8828-abc7e372a09f)
![image](https://github.com/user-attachments/assets/ce06818d-aacf-4bc2-af4f-2a3ba4798015)

evertwined villages section starts visible (all tasks are optional)
![image](https://github.com/user-attachments/assets/25ade386-bce8-460b-87ce-c77d2e5c5726) 
![image](https://github.com/user-attachments/assets/67dffc01-b3a7-4676-ace5-e67746baf991)

thats everything I've found for now. 

please make a new instance from this branch before making further changes so that other changes you may have in other instances ar not carried over.

re-request review once changes are completed

Jonah-Hansen commented:
also wtf is this? damages the player as soon as they spawn in 
![image](https://github.com/user-attachments/assets/e0df727c-566d-4fdd-aec7-bb25f368512a)

RichPrune commented on src/overrides/scripts/inventory-mod/block6.zs:
I think it automatically added them to the bestiary pr because prs compare branches to branches. I didn't intend to though. I'll make a new branch in the future for feature/inventorymod changes

Jonah-Hansen replied:
please update this PR description with a list of all changes this pr will make compared to the base branch. use - to make bullet points. 

example: 

- adds more tags to inventory mod tags
- add beastiary questbook page
- replace questbook item with dormant eye

RichPrune replied:
kk

RichPrune commented:
Removed the some mob icons rp from paxi and added to rpoverrides
Also added Journey map icons which was missing by accident( both in via manifest/modlist)
Deleted execute meteor quest, glowing eye quests in temp
Reenabled bestiary toasts
Removed tasks for bestiary and legend quests and added xp rewards
Added visit dim everbright and everdawn task to evertwined villages
Added checkmark to spawn blockers quest
Set temps to order 1
Added forest to forests category tag

Jonah-Hansen commented on src/manifest.json:
do not unnecesarily change formatting

(files array should be tabbed in 1 more. to do so you can highlight all lines and press 'tab' to indent all lines)

when reviewing changes to files it should only show red/green on the lines you intended to change. if you have a giant block of red/green like this you know you messed something up

Jonah-Hansen commented on src/modlist.html:
do not unnecesarily change formatting

(the lines should be tabbed one less. to do so you can highlight all lines and press 'shift + tab' to un-indent all lines)

Jonah-Hansen commented on src/overrides/config/ftbquests/quests/chapters/temps.snbt:
```diff
@@ -3810,6 +3933,110 @@
+			x: 0.0d
+			y: -1.5d
+		}
 	]
```
remove added biome particles

Jonah-Hansen commented on src/overrides/config/resourcepackoverrides.json:
```diff
@@ -83,7 +83,9 @@
+      "choccompatv1.3",
+      "Some Mobs Icons 1.5",
+      "JourneyMap Extended Mob Icons v2.0"
     ],
```
these should also be added to the "default_packs" list above. preferably put them in a reasonable order (put xaero icons with the other xaero icon packs)

these should also be `"file/resourcepack.zip"` to use the zip from resourcepacks folder instead of unzipped paxi folder

Jonah-Hansen commented:
closing as features have been re-implemented in #916 and #918. I left out the inventory mod scripts as there are invalid references and they should not be implemented on develop yet. 

if you want to push them to git, make a branch based off the inventory-mod-develop branch and open a PR there
